src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9666,8 +9666,7 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
     // set position before any AI calls/assistance
     //if (GetTypeId() == TYPEID_UNIT)
     //    ToCreature()->SetCombatStartPosition(GetPositionX(), GetPositionY(), GetPositionZ());
-
-    if (creature && !IsControllableGuardian())
+    if (creature && !IsControlledByPlayer())
```

Comment:
`if (creature && !IsControllableGuardian() && !IsControlledByPlayer())`
We want that check about `IsControllableGuardian` as it should only affect controllable guardians.

Reply:
that way a controllable guardian owned by an npc never gon have u on threatlist unless u attack him. this fucks up errythang else

Reply:
Okay.